### ISSUE TITLE
multithreaded decoding of Big_Buck_Bunny_1080_10s_30MB.mp4 fails

### ISSUE THREAD

**Issue by jrmuizel** [cisco/openh264#3487]
Steps to reproduce:
```
$ wget https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/1080/Big_Buck_Bunny_1080_10s_30MB.mp4
$ ffmpeg -i Big_Buck_Bunny_1080_10s_30MB.mp4 -map 0:0 -vcodec copy -an Big_Buck_Bunny_1080_10s_30MB.264
```
apply
```patch
diff --git a/codec/console/dec/src/h264dec.cpp b/codec/console/dec/src/h264dec.cpp
index 59365844..4338bbce 100644
--- a/codec/console/dec/src/h264dec.cpp
+++ b/codec/console/dec/src/h264dec.cpp
@@ -617,21 +617,21 @@ int32_t main (int32_t iArgC, char* pArgV[]) {


   if (WelsCreateDecoder (&pDecoder)  || (NULL == pDecoder)) {
     printf ("Create Decoder failed.\n");
     return 1;
   }
   if (iLevelSetting >= 0) {
     pDecoder->SetOption (DECODER_OPTION_TRACE_LEVEL, &iLevelSetting);
   }

-  int32_t iThreadCount = 0;
+  int32_t iThreadCount = 3;
   pDecoder->SetOption (DECODER_OPTION_NUM_OF_THREADS, &iThreadCount);

   if (pDecoder->Initialize (&sDecParam)) {
     printf ("Decoder initialization failed.\n");
     return 1;
   }


   int32_t iWidth = 0;
   int32_t iHeight = 0;
```
```
$ ./h264dec Big_Buck_Bunny_1080_10s_30MB.264
```

This gives:

`[OpenH264] this = 0x0x7fe2ce808800, Error:DecodeCurrentAccessUnit()::::::PrefetchPic ERROR, pSps->iNumRefFrames:16`

and the decoder then hangs.

**Comment by xiaotiansf** [comment]
@jrmuizel I will look at the issue.

**Comment by xiaotiansf** [comment]
@jrmuizel Just let you know that I am making progress.

**Comment by huili2** [comment]
any update on this issue?

**Comment by xiaotiansf** [comment]
This has been fixed at commit 0819f28.
